src/main/resources/XWiki/Macros/Team.xml

```diff
@@ -0,0 +1,1491 @@
+    </property>
+    <property>
+      <code>{{velocity}}
+#set($discard = $xwiki.ssx.use('xwiki:XWiki.Macros.Team'))
+#set($discard = $xwiki.jsx.use('xwiki:XWiki.Macros.Team'))
+#set($pictureList = {})
+#macro(avatarInitials $name $size $letterAvatarBgColor $letterAvatarFontColor)
+  #set($escapedSize = $escapetool.xml($size))
+  #set($escapedLetterAvatarFontColor = $escapetool.xml($letterAvatarFontColor))
+  #set($escapedLetterAvatarBgColor = $escapetool.xml($letterAvatarBgColor))
+  &lt;span
+    class="xwikiteam-avatar xwikiteam-avatar-initials"
+    style="height:${escapedSize}px; width:${escapedSize}px; border-radius: ${escapedSize}px; background-color: $escapedLetterAvatarBgColor"
+  &gt;
+    &lt;span class="xwikiteam-avatar-initials-inner"&gt;
+      &lt;span class="xwikiteam-avatar-initials-letters" style="color: $escapedLetterAvatarFontColor"&gt;
+        #foreach($firstLetter in $name.toUpperCase().split("[\s-]"))$escapetool.xml($firstLetter.charAt(0))#end
+      &lt;/span&gt;
+    &lt;/span&gt;
+  &lt;/span&gt;
+#end
+#macro(avatar $username $size $disableLetterAvatars $letterAvatarBgColor $letterAvatarFontColor)
+  #set($displayUser = $xwiki.getUserName($username, false))
+  #set($escapedDisplayUser = $escapetool.xml($displayUser))
+  #set($fontSize = $size / 3)
+  &lt;li
+    class="xwikiteam-user"
+    data-name-without-accents="${util.clearAccents($escapedDisplayUser.toLowerCase())}"
+    title="$escapedDisplayUser"
+    data-username="${escapetool.xml($username)}"
+    style="font-size: ${fontSize}px"
+  &gt;
+    &lt;a href="$url"&gt;
+      #getUserAvatarURL($username $return $size)
+      #if(!$return.specified &amp;&amp; !$disableLetterAvatars)
+        #avatarInitials($displayUser $size $letterAvatarBgColor $letterAvatarFontColor)
+      #else
+        &lt;img
+          class='xwikiteam-avatar'
+          src='$escapetool.xml($return.url)'
+          alt='$escapedDisplayUser'
+          title='$escapedDisplayUser'
+          width='$size'
+          height='$size'
+        /&gt;
+      #end
+      &lt;span class='xwikiteam-username'&gt;$escapedDisplayUser&lt;/span&gt;
+    &lt;/a&gt;
+  &lt;/li&gt;
+#end
+#set($tag = $wikimacro.parameters.tag)
+#set($size = $wikimacro.parameters.size)
+#set($letterAvatarBgColor = $wikimacro.parameters.letterAvatarBgColor)
+#set($letterAvatarFontColor = $wikimacro.parameters.letterAvatarFontColor)
+#set($requireExternalAuth = $wikimacro.parameters.requireExternalAuth)
+#set($showUsernames = $wikimacro.parameters.showUsernames)
+#set($disableTools = $wikimacro.parameters.disableTools)
+#set($disableLetterAvatars = $wikimacro.parameters.disableLetterAvatars)
+#set($limit = $wikimacro.parameters.limit)
+#set($configObject = $xwiki.getDocument("XWiki.Macros.TeamConfiguration").getObject("XWiki.Macros.TeamConfigurationClass"))
+#if("$!{size}" == "")
+  #set($size = $configObject.getValue("size"))
+#end
+#set($size = $escapetool.xml($size))
+#if("$!{size}" == "")
+  #set($size = 60)
```

Comment:
To be checked, but I think this line (as well as those below) is not reached now that you have the default value specified in the macro parameter descriptor. It's not a problem for the size parameter, but fort the rest, if you need to rely on external configuration (i.e. control de default values from the administration) it will be a problem because the value configured in the administration is not going to be used. This approach is not common and in fact this is the first macro I know that comes with a configuration section to control the default parameter values. The problem is that there's no perfect solution ATM, I think:

* the parameter descriptor should specify the default values so that the WYSIWYG editor doesn't output ``param=value`` when inserting a macro and using the default parameter value
* you need the default values outside of macro definition which is read-only for licensed extensions, if you want to allow the user to change them

This deserves a discussion on the forum to see what is the best way to make the default macro parameter values configurable.

Reply:
I'd argue for:

Reply:
The problem is that if you define the default values outside of the macro parameter descriptor then anyone relying on the macro parameter descriptor (e.g. the WYSIWYG editor) to read the default values will get the wrong information. E.g. if some tool generates documentation for macros and shows the default values for macro parameters, then that automatically generated documentation will be wrong for the teams macro. I still think this needs a discussion.

Reply:
https://forum.xwiki.org/t/handling-configurable-default-parameters-in-a-macro/10094

Reply:
@mflorea would using advanced parameters as suggested by Vincent here https://forum.xwiki.org/t/handling-configurable-default-parameters-in-a-macro/10094/2 be an acceptable solution? It requires using Java code. Or should we go with something simpler and make this macro less configurable by not allowing users to use parameters or admins to set defaults? WDYT?

Reply:
I decided to remove the configuration class for now, we can always make the team macro configurable in the future if needed